### PR TITLE
#4509: enable selecting/unselecting all when creating new authentication file from subscriptions in Intellij Idea

### PR DESCRIPTION
**Pull request by wangmingliang-ms**
refer https://github.com/microsoft/azure-tools-for-java/issues/4509

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/utility/JTableUtils.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                column.setHeaderValue(false);
+            } else {
+                column.setHeaderValue(null);
+            }
+            this.table.getTableHeader().repaint();
+        }
+
+        private void syncTableModelToHeaderValue(boolean headerValue) {
+            final int rowCount = table.getModel().getRowCount();
+            this.updatingHeaderValue = true;
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/utility/JTableUtils.java:105]: change to use `synchronized` or lock ?

Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/utility/JTableUtils.java:105]: There should not be synchronize issues

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/utility/JTableUtils.java [hunk 2]
```diff
@@ -84,6 +72,27 @@ public void mouseClicked(MouseEvent e) {
             }
         }
 
+        private void updateTableModel(final boolean headerValue) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/utility/JTableUtils.java:75]: move private methods to last

**Comment by andxu** [comment]
Can we use JTableUtils  for solving other select all subscription tables?

**Comment by wangmingliang-ms** [comment]
> Can we use JTableUtils for solving other select all subscription tables?
@andxu yes, we can

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/utility/JTableUtils.java (hunk 2 above)
Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/utility/JTableUtils.java]: fixed

**Comment by wangmingliang-ms** [comment]
@andxu @Flanker32  do you have any concerns for this PR? can this PR be merged now?

**Review by andxu — APPROVED** [review]

**Pull request merged**